openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1352,9 +1352,10 @@ This specification defines the following Credential Issuer Metadata parameters:
-  * `proof_types`: OPTIONAL. Array of case sensitive strings, each representing a `proof_type` that the Credential Issuer supports, as defined in (#credential_request), one of which MUST be used in the Credential Request. If this array is non-empty and present, the Credential Issuer requires proof of possession of the cryptographic key material. If the parameter is omitted or the array is empty, the Credential Issuer does not require proof of possession of the cryptographic key material.
+  * `cryptographic_binding_methods_supported`: OPTIONAL. Array of case sensitive strings that identify the representation of the cryptographic key material that the issued Credential is bound to, as defined in (#credential-binding). Support for keys in JWK format [@!RFC7517] is indicated by the value `jwk`. Support for keys expressed as a COSE Key object [@!RFC8152] (for example, used in [@!ISO.18013-5]) is indicated by the value `cose_key`. When the Cryptographic Binding Method is a DID, valid values are a `did:` prefix followed by a method-name using a syntax as defined in Section 3.1 of [@!DID-Core], but without a `:`and method-specific-id. For example, support for the DID method with a method-name "example" would be represented by `did:example`. Support for all DID methods listed in Section 13 of [@DID_Specification_Registries] is indicated by sending a DID without any method-name.
+  * `credential_signing_alg_values_supported`: OPTIONAL. Array of case sensitive strings that identify the algorithms that the Issuer uses to sign the issued Credential. Algorithm names used are determined by the Credential format and are defined in (#format_profiles).
+  * `proofs_supported`: OPTIONAL. Object that describes specifics of the key proof(s) that the Credential Issuer supports. This object contains a list of name/value pairs, where each name is a unique identifier of the supported proof type being described. This identifier is used in the Credential Request defined in (#credential_request) to communicate to the Wallet specifics about the proof of possession of the cryptographic key material. A list of valid key proof type values defined by this specification can be found in (#proof_types). The value in the name/value pair is an object that contains metadata about the key proof and contains the following parameters defined by this specification:
+    * `proof_signing_alg_values_supported`: REQUIRED. Array of case sensitive strings that identify the algorithms that the Issuer uses to sign the proof. Algorithm names used SHOULD be one of those defined in [@IANA.JOSE.ALGS], [@LD_Suite_Registry] or [@!ISO.18013-5].
```

Comment:
If we're already making this an object per proof_type, i think it might be useful to define the structure of the object in the proof key types section. For `cwt` and `jwt` ,`proof_signing_alg_values_supported` makes sense. However for data integrity proofs, you can have different types (`DataIntegrityProof`, `Ed25519Signature2020`), but also for `DataIntegrityProof` you have `cryptosuites`.

E.g. for `credential_signing_alg_values_supported` it mentions "Algorithm names used are determined by the Credential format and are defined in (#format_profiles)", so maybe we can make this:

```suggestion
  * `proofs_supported`: OPTIONAL. Object that describes specifics of the key proof(s) that the Credential Issuer supports. This object contains a list of name/value pairs, where each name is a unique identifier of the supported proof type being described. This identifier is used in the Credential Request defined in (#credential_request) to communicate to the Wallet specifics about the proof of possession of the cryptographic key material. A list of valid key proof type values defined by this specification can be found in (#proof_types). The value in the name/value pair is an object of which the value is determined by the Key Proof Type and are defined in (#key-proof-types).
```

Reply:
If we do want to keep `proof_signing_alg_values_supported`, I think it at least makes sense to change

> Algorithm names used SHOULD be one of those defined in [@IANA.JOSE.ALGS], [@LD_Suite_Registry] or [@!ISO.18013-5].

to 

> Algorithm names used are determined by the Key Proof Type and are defined in (#key_proof_types).

And add the following sentence to each of the key proof types (with the correct spec):

> Cryptographic algorithm names used in the `proof_signing_alg_values_supported` parameter SHOULD be one of those defined in [@IANA.JOSE.ALGS].


This makes it consistent with `credential_signing_alg_values_supported`

Reply:
this differentiation is on purpose because the algs in the proof types are dependent on the proof type (jwt, jdp, cwt) ad not the credential typ, so it is not accurate to point to the credential profiles for the algs used in the proofs

Reply:
> so it is not accurate to point to the credential profiles for the algs

I don't mean for them to point to the credential profiles, but to the Key Proof Types section where the actual proof types are defined: https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-wg-draft.html#section-7.2.1

So Section 7.2.1.1 will mention:

> Cryptographic algorithm names used in the `proof_signing_alg_values_supported` parameter for this supported proof type SHOULD be one of those defined in [@IANA.JOSE.ALGS].

And Section 7.2.1.2 will mention:

> Cryptographic algorithm names used in the `proof_signing_alg_values_supported` parameter for this supported proof type SHOULD be one of those defined in [@LD_Suite_Registry].

Reply:
ok. that makes sense - thank you for clarifying. Happy to make this change if there is an agreement on the WG call (just to prevent the double work of potentially reverting.)

Reply:
Discussed in WG call on 31 Jan (NZ). People in the call leaned towards moving the proof type specific algs to the key proof type section.

Reply:
changes applied